nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/service/flow.service.ts

```diff
@@ -118,6 +118,11 @@ export class FlowService implements PropertyDescriptorRetriever {
         return this.httpClient.get(`${FlowService.API}/process-groups/${id}`);
     }
 
+    getProcessGroupWithContent(id: string): Observable<any> {
+        const params = new HttpParams().set('includeContent', true);
+        return this.httpClient.get(`${FlowService.API}/process-groups/${id}`, { params: params });
+    }
```

Comment:
This change is necessary for this feature to ensure that when moving a Controller Service it remains in scope for all components currently referencing it. However, I have some concerns about fetching the entire flow from the current Process Group when the Move dialog is opened. I wonder if we should consider a purpose built endpoint for does exactly what we need instead of returning everything. We've certainly seen some large flows over the years where I worry this could be problematic.

CC @exceptionfactory

Reply:
I've created a new API endpoint, move-options, to get the information needed for the process groups. This shifts all the logic to the API, and greatly simplifies the UI form.

Reply:
I agree. This refactoring has made the UI much simpler.